src/Core__Array.resi

```diff
@@ -125,17 +691,149 @@ let reduceRight: (array<'a>, 'b, ('b, 'a) => 'b) => 'b
 */
 let reduceRightWithIndex: (array<'a>, 'b, ('b, 'a, int) => 'b) => 'b
 
-@send external some: (array<'a>, 'a => bool) => bool = "some"
-@send external someWithIndex: (array<'a>, ('a, int) => bool) => bool = "some"
-@get_index external get: (array<'a>, int) => option<'a> = ""
-@set_index external set: (array<'a>, int, 'a) => unit = ""
+/**
+`some(array, checker)` returns true if running the provided `checker` function on any element in `array` returns true.
+
+See [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) on MDN.
+
+## Examples
+```rescript
+let array = ["Hello", "Hi", "Good bye"]
+
+Console.log(array->Array.some(greeting => greeting === "Hello")) // true
+```
+*/
+@send
+external some: (array<'a>, 'a => bool) => bool = "some"
+
+/**
+`someWithIndex(array, checker)` returns true if running the provided `checker` function on any element in `array` returns true.
+
+See [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) on MDN.
+
+## Examples
+```rescript
+let array = ["Hello", "Hi", "Good bye"]
+
+Console.log(array->Array.someWithIndex((greeting, index) => greeting === "Hello" && index === 0)) // true
+```
+*/
+@send
+external someWithIndex: (array<'a>, ('a, int) => bool) => bool = "some"
+
+/**
+`get(array, index)` returns the element at `index` of `array`.
+
+Returns `None` if the index does not exist in the array. Equivalent to doing `array[index]` in JavaScript.
+
+## Examples
+```rescript
+let array = ["Hello", "Hi", "Good bye"]
+
+array->Array.get(0) == Some("Hello") // true
+```
+*/
+@get_index
+external get: (array<'a>, int) => option<'a> = ""
+
+/**
+`set(array, index, item)` sets the provided `item` at `index` of `array`.
+
+Beware this will *mutate* the array.
+
+## Examples
+```rescript
+let array = ["Hello", "Hi", "Good bye"]
+array->Array.set(1, "Hello")
+
+Console.log(array[1]) // "Hello"
+```
+*/
+@set_index
+external set: (array<'a>, int, 'a) => unit = ""
 @get_index external getSymbol: (array<'a>, Core__Symbol.t) => option<'b> = ""
 @get_index external getSymbolUnsafe: (array<'a>, Core__Symbol.t) => 'b = ""
 @set_index external setSymbol: (array<'a>, Core__Symbol.t, 'b) => unit = ""
+
+/**
+`getUnsafe(array, index)` returns the element at `index` of `array`.
+
+This is _unsafe_, meaning it will fail with an exception if `index` does not exist in `array`.
+
+## Exceptions
+
+- `Not_found`: If the provided `index` does not exist in `array`.
+
+## Examples
+```rescript
+let array = ["Hello", "Hi", "Good bye"]
+
+Console.log(array->Array.getUnsafe(0)) // "Hello"
+Console.log(array->Array.getUnsafe(3)) // Fails and raises exception
+```
+*/
+@raises(Not_found)
 external getUnsafe: (array<'a>, int) => 'a = "%array_unsafe_get"
+
+// TODO: Docs. Is there any practical difference from `set`?
 external setUnsafe: (array<'a>, int, 'a) => unit = "%array_unsafe_set"
```

Comment:
Yes. `set` does bounds checking. If `index` is outside the array, `set` will raise an exception while `setUnsafe` will set the element at `index` and effectively also insert `undefined` items between `index` and the previous end of the array.

Reply:
Had no idea. Thanks!